### PR TITLE
#3232: Providing Rename Thread and etc. for CallStack View

### PR DESCRIPTION
**Pull request by foralost**
Adding `Follow in Threads` and `Rename Thread` option to context menu.
image

Solves #3232

### CODE REVIEW

**Comment by foralost** [comment]
I would need help regarding switching to `Threads` view, because I have found that `Disassembly` window is quite supported, but that one is not.

**Comment by foralost** [comment]
And also the CI says that my new-added files are nonconforming, but locally it says otherwise:
image

**Comment by mrexodia** [comment]
Most likely the reason is your CRLF settings. To fix run: `git config --global core.crlf false`

**Comment by foralost** [comment]
@mrexodia  nah, it did not work :< , git diff shows nothing with enabled whitespaces... im using CRLF.
(I have reset branch and committed again, maybe setting everything from ground up will do the trick).

I will try to investigate the CPP solution and study the disasm command on the weekend to find out how the Dissasembly window is called. Probably there will be a new command to switch to the Threads window, but will see.

**Comment by mrexodia** [comment]
The core.autocrlf option automatically converts the whitespaces in the commit. The files still stay CRLF on your local checkout. I’ll take a look tomorrow, but this is the only reason I can think of…

**Comment by mrexodia** [comment]
Just confirmed, the newly-created files `ThreadGUIUtils.cpp` and `ThreadGUIUtils.h` use LF line endings.

**Comment by mrexodia** [comment]
I think in general it is not necessary to create new files for this functionality. Simply copying the code in both places will work just fine.

**Comment by mrexodia** [comment]
Ah I just remembered there is a `SimpleInputBox` function in `MiscUtil.h` that does pretty much exactly what you want.

**Comment by foralost** [comment]
@mrexodia those god damn newlines... 

Ok, then on weekend I will continue progress, thanks!

**Comment by foralost** [comment]
Added the `showthreadid` command, which switches to the `Threads` view (with also implementing new messages for switching to this view). Also used in the new context menu.

Used `MiscUtils.h` as suggested for both locations.

If everythings works ok, I will fix alphabetical order in function names etc.  Plz test.

File: src/bridge/bridgemain.h [hunk 1]
```diff
@@ -1149,6 +1149,7 @@ typedef enum
     GUI_DISASSEMBLY,
     GUI_DUMP,
     GUI_STACK,
+    GUI_THREADS,
```
Comment (mrexodia) [inline src/bridge/bridgemain.h:1152]: You cannot insert enums for binary compatibility. Please append it to the end

File: src/bridge/bridgemain.h [hunk 2]
```diff
@@ -1281,6 +1282,8 @@ typedef enum
     GUI_SAVE_LOG,                   // param1=const char* file name,param2=unused
     GUI_REDIRECT_LOG,               // param1=const char* file name,param2=unused
     GUI_STOP_REDIRECT_LOG,          // param1=unused,               param2=unused
+    GUI_SELECT_THREAD_BY_ID,        // param1=ThreadID
```
Comment (mrexodia) [inline src/bridge/bridgemain.h:1285]: I think the existing “GuiSelectionSet” message can be used for this

File: src/dbg/commands/cmd-gui.cpp [hunk 3]
```diff
@@ -8,6 +8,36 @@
 #include "value.h"
 #include "variable.h"
 
+bool cbShowThreadId(int argc, char* argv[])
```
Comment (mrexodia) [inline src/dbg/commands/cmd-gui.cpp:11]: This should be merged with the existing command: https://github.com/x64dbg/x64dbg/blob/development/src/dbg/commands/cmd-thread-control.cpp#L49

**Review by mrexodia — CHANGES_REQUESTED** [review]
In general I think the selection api messages can be reused, otherwise looks mostly fine. Thanks!

File: src/bridge/bridgemain.h (hunk 1 above)
Comment (foralost) [reply src/bridge/bridgemain.h]: Appended to the end.

**Comment by mrexodia** [comment]
Thanks for fixing the suggestions, wasn't home this weekend but will try to merge it asap

**Comment by foralost** [comment]
The thread ID should be provided in HEX or in DEC? Cause I saw in  #3271 that there [ToHex is used](https://github.com/x64dbg/x64dbg/pull/3271/files#diff-8d202accfd12cde659d9fd3c54531ec206ff0672e1e9973bdfa39a5d71a92603R267).

**Comment by mrexodia** [comment]
There is a GUI option to choose, but I think we’ll be removing it. I think there is a helper function in StringUtils to format it automatically though…

**Comment by mrexodia** [comment]
Thank you!

**Pull request merged**